Add op: masked_select

manuelcandales opened the pull request:
Differential Revision: D65497030

jackzhxng reviewed (commented):
Should we also make this core aten?

manuelcandales committed:
Add op: masked_select (#6670)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/6670

Reviewed By: dvorjackz

Differential Revision: D65497030
